src/Rules/Drupal/EntityQuery/EntityQueryHasAccessCheckRule.php

```diff
@@ -40,7 +40,7 @@ public function processNode(Node $node, Scope $scope): array
 
         return [
             RuleErrorBuilder::message(
-                'Missing explicit access check on entity query.'
+                'Relying on entity queries to check access by default is deprecated in drupal:9.2.0 and an error will be thrown from drupal:10.0.0. Call \Drupal\Core\Entity\Query\QueryInterface::accessCheck() with TRUE or FALSE to specify whether access should be checked. See https://www.drupal.org/node/3201242.'
             )->tip('See https://www.drupal.org/node/3201242')->build(),
         ];
```

Comment:
I wanted to add a logic branch here. If Drupal < 10.0.x show deprecated message, otherwise report as an error.

Reply:
This shouldn't block the merge